tests/test_global_fail.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+
+from pathlib import Path
+
+import biorbd
+from casadi import vertcat, MX
+
+from bioptim import (
+    Node,
+    OptimalControlProgram,
+    DynamicsTypeOption,
+    DynamicsType,
+    ObjectiveOption,
+    Objective,
+    ConstraintList,
+    BoundsOption,
+    QAndQDotBounds,
+    InitialGuessOption,
+    ShowResult,
+    OdeSolver,
+)
+
+
+def custom_func_align_markers(ocp, nlp, t, x, u, p, first_marker_idx, second_marker_idx):
+    nq = nlp.shape["q"]
+    val = []
+    markers = biorbd.to_casadi_func("markers", nlp.model.markers, nlp.q)
+    for v in x:
+        q = v[:nq]
+        first_marker = markers(q)[:, first_marker_idx]
+        second_marker = markers(q)[:, second_marker_idx]
+        val = vertcat(val, first_marker - second_marker)
+    return MX(val - 1), val, MX(val + 1)
+
+
+def prepare_ocp(biorbd_model_path, ode_solver=OdeSolver.RK):
```

Comment:
That seems an overkill test. In the end, it is not clear what you are trying to test... For instance, if objective_functions is not expected to fail, then do not add it, it is not a required element of OCP. At the end, all this prepare_ocp function should be able to be wrote in 4 lines max (which raises the question why to put this prepare_ocp in an function in the first place instead of directly in the test.

Same thing for the custom_func_align_markers, does this make sense to test the failure? Otherwise it should not appear